### DISCUSSION TITLE
Support vcpkg integrate bash on Windows

### DISCUSSION THREAD

**Discussion by linquize** [microsoft/vcpkg#19143] [Ideas]
To support `vcpkg integrate bash` on Windows as well.

based on existing code, modify something
1. For `home_path`, Get User's directory instead of `%HOME%`
2. Write `.bashrc` with '/' style path

**Comment by MVoz** [comment]
bash/sh msys/mingw32(64) or wsl? 

msys/mingw
`downloads\tools\msys2\msys64` *.ini -->> `MSYS2_PATH_TYPE=inherit`
`env = MSYS2_PATH_TYPE = inherit`

test
cmd.exe `set MSYS2_PATH_TYPE=inherit` && ..\msys2\msys64\usr\bin\ `sh.exe -lc "realpath ~"`

**Comment by linquize** [comment]
It doesn't matter you use what bash. When you write something to `.bashrc` in User's home directory, any bash will read that file.

WSL runs linux version of vcpkg so it is already implemented

**Comment by heydojo** [comment]
Git bash? https://initgeek.com/how-to-use-git-git-bash-and-github/
WSL cross compilation?

@linquize would you mind please clarifying what it is you mean please?
vcpkg already supports invocation via Windows built in command processor (cmd.exe) and also powershell.
bash isn't a native shell on Windows, so I don't know why or how it would be possible to support that officially, unless you could maybe be more specific about a port of bash for Windows which exists that you have tried and is actively maintained.
The only one which springs to mind for me is git bash but I don't use it myself.

**Comment by linquize** [comment]
Simply append one line to the end of `~/.bashrc`, which is `C:\Users\user\.bashrc`

No need to find where win32 bash is located. 

Unrelated to WSL.

**Comment by smacdonald-kai** [comment]
Hoping this re-opens the issue.

Reproduction steps:
- start Windows Sandbox
- install Git for Windows ( https://gitforwindows.org/ )
- start a Git for Windows bash instance
- install vcpkg

Observe the following result:
```
WDAGUtilityAccount@c9bf4b53-778e-42d6-89b6-b62ebe017eab MINGW64 ~/qwe
$ vcpkg --version
vcpkg package management program version 2025-12-16-44bb3ce006467fc13ba37ca099f64077b8bbf84d

See LICENSE.txt for license information.

WDAGUtilityAccount@c9bf4b53-778e-42d6-89b6-b62ebe017eab MINGW64 ~/qwe
$ vcpkg integrate bash
error: vcpkg integrate bash is non-Windows-only and not supported on this system.

WDAGUtilityAccount@c9bf4b53-778e-42d6-89b6-b62ebe017eab MINGW64 ~/qwe
$ bash --version
GNU bash, version 5.2.37(1)-release (x86_64-pc-msys)
Copyright (C) 2022 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

This is free software; you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
```

Expected result is that vcpkg notices it is running inside of a bash instance, and does the the integration.
